src/Microsoft.DotNet.ProjectModel/Files/ProjectFilesCollection.cs

```diff
@@ -79,13 +79,16 @@ internal void EnsureInitialized()
             _namedResources = NamedResourceReader.ReadNamedResources(_rawProject, _projectFilePath);
 
             // Files to be packed along with the project
-            var packIncludeJson = _rawProject.ValueAsJsonObject(PackIncludePropertyName);
+            var packIncludeJson = _rawProject.Value<JToken>(PackIncludePropertyName) as JObject;
             if (packIncludeJson != null)
             {
-                _packInclude = packIncludeJson
-                    .Keys
-                    .Select(k => new PackIncludeEntry(k, packIncludeJson.Value(k)))
-                    .ToList();
+                var packIncludeEntries = new List<PackIncludeEntry>();
+                foreach (var token in packIncludeJson)
+                {
+                    packIncludeEntries.Add(new PackIncludeEntry(token.Key, token.Value));
+                }
+
+                _packInclude = packIncludeEntries;
```

Comment:
Easier said than done. We would need to add a bunch of dependencies to be able to do the select on the JObject directly. I preferred not to do that.